lua/gatekeeper/staticlib.lua

```diff
@@ -74,6 +87,48 @@ function alloc_an_lcore (numa_table)
 	return lcore_t[1]
 end
 
+function alloc_lcores_evenly_from_all_numa_nodes (numa_table, n,
+		fixed_lcores_per_numa)
+	local num_numa_nodes = count_numa_nodes(numa_table)
+	local q = n / num_numa_nodes
+	local r = n % num_numa_nodes
+	local i = 0
+	local res = {}
+	for numa, lcores in all_ipairs(numa_table) do
+		local lcores_needed = q + ((i < r) and 1 or 0)
+		if lcores_needed > 0 then
+			lcores_needed = lcores_needed + fixed_lcores_per_numa
+		else
+			break
+		end
+		if #lcores >= lcores_needed then
+			res[numa] = alloc_lcores_at_numa(numa_table, numa, lcores_needed)
+		else
+			error("There is not enough lcores");
+		end
+		i = i + 1
+	end
+	res["__net_conf"] = numa_table["__net_conf"]
+	return res
+end
+
+local function merge_arrays (a, b)
+	for i, v in ipairs(b) do
+		table.insert(a, v)
+	end
+	return a
+end
+
+function convert_numa_table_to_array (numa_table)
+	local res = {}
+	for numa, lcores in all_ipairs(numa_table) do
+		if numa ~= "__net_conf" then
```

Comment:
In fact, we don't need to check for field `__net_conf` here because we are already using `all_pairs()`. It was my fault.